pandas/core/generic.py

```diff
@@ -1060,8 +1057,13 @@ def f(x):
             baxis = self._get_block_manager_axis(axis)
             if level is not None:
                 level = self.axes[axis]._get_level_number(level)
-            result._data = result._data.rename_axis(f, axis=baxis, copy=copy,
-                                                    level=level)
+
+            # TODO: we already did a copy above, can we avoid doing again?
```

Comment:
I think this makes it more confusing here, reaching into internals to do this.

I would rather move _transform_index and items_overlap_with_suffix from internals to Index.base (or appropriate), these are just renaming / transform helpers for an index.

Then this would be much cleaner.

Reply:
> I think this makes it more confusing here, reaching into internals to do this.

At least add_suffix and add_prefix are clear wins.  I can de-`_data` this a little bit by using `NDFrame.set_axis`.  I'll take a look at transform_index and items_overlap_with_suffix, see if there is a more reasonable place for them.

> I would rather [...]

I agree, am working along similar lines.